src/broyden.jl

```diff
@@ -0,0 +1,170 @@
+# based on 
+# A QUASI-NEWTON METHOD FOR SOLVING SMALL NONLINEAR SYSTEMS OF ALGEBRAIC EQUATIONS
+# JEFFREY MARTIN
+# http://web.mit.edu/jmartin3/Public/Project.pdf
+macro broydentrace(stepnorm)
```

Comment:
This is how it used to be done in Optim, and I'm pretty sure NLsolve just mimicked it (we have gone away from this in Optim).